Core/Src/main.c

```diff
@@ -66,6 +68,25 @@ int main(void)
 {
   /* USER CODE BEGIN 1 */
 
+  // MAX_ADC_VAL = 2^10 - 1
+  const uint16_t MAX_ADC_VAL = 0b1111111111;
+  const int TIMER_PERIOD = 64000;
```

Comment:
int is perfectly fine here, but a uint16_t would do better here in this case bec it is explicity in the sense that the value is greater than 0, and we know the exact memory size (2 bytes).